server/src/com/cloud/network/guru/PublicNetworkGuru.java

```diff
@@ -115,7 +115,11 @@ protected PublicNetworkGuru() {
     protected void getIp(NicProfile nic, DataCenter dc, VirtualMachineProfile vm, Network network) throws InsufficientVirtualNetworkCapacityException,
         InsufficientAddressCapacityException, ConcurrentOperationException {
         if (nic.getIPv4Address() == null) {
-            PublicIp ip = _ipAddrMgr.assignPublicIpAddress(dc.getId(), null, vm.getOwner(), VlanType.VirtualNetwork, null, null, false);
+            boolean forSystemVms = false;
+            if (vm.getType().equals(VirtualMachine.Type.ConsoleProxy) || vm.getType().equals(VirtualMachine.Type.SecondaryStorageVm)) {
+                forSystemVms = true;
```

Comment:
Don't we need to check the value of SystemVmPublicIpReservationModeStrictness parameter before assigning the value as true for SSVM and CPVM?

Reply:
Could be another alternative, actually logic is done on `IpAddressManagerImpl.fetchNewPublicIp` method using `forSystemVms` and `SystemVmPublicIpReservationModeStrictness` value.

Reply:
@nvazquez, I just went with the flow and found out there already exists a column name ("is_system") in user_ip_address table. Can you please see if you can make use of that instead of creating a new one ("forsystemvms").

Reply:
@nitin-maharana sorry for late response, been busy with other tasks. I checked usage of is_system column and are used for static nat or load balancing purposes. Best option should be including this new column.